libraries/Filter/HarmonicNotchFilter.cpp

```diff
@@ -230,25 +331,14 @@ void HarmonicNotchFilter<T>::update(float center_freq_hz)
     // update all of the filters using the new center frequency and existing A & Q
     for (uint8_t i = 0; i < HNF_MAX_HARMONICS && _num_enabled_filters < _num_filters; i++) {
         if ((1U<<i) & _harmonics) {
-            const float notch_center = center_freq_hz * (i+1);
+            const float notch_center = center_freq_hz;
+            const uint8_t harmonic_mul = i+1;
             if (_composite_notches != 2) {
-                // only enable the filter if its center frequency is below the nyquist frequency
-                if (notch_center < nyquist_limit) {
-                    _filters[_num_enabled_filters++].init_with_A_and_Q(_sample_freq_hz, notch_center, _A, _Q);
-                }
+                set_center_frequency(_num_enabled_filters++, notch_center, 1.0, harmonic_mul);
             }
             if (_composite_notches > 1) {
-                float notch_center_double;
-                // only enable the filter if its center frequency is below the nyquist frequency
-                notch_center_double = notch_center * (1.0 - _notch_spread);
-                if (notch_center_double < nyquist_limit) {
```

Comment:
The reason for checking the nyquist limit here was so that if it was disabled _num_enabled_filters would not advance and you wouldn't get a dead filter in your list of enabled filters.

Reply:
but that is a transient condition based on current source frequency, we don't want the number of filters to change dynamically as that will mean we'll move which filter object is used which will cause both filter glitches and more re-init